compiler/src/dotty/tools/dotc/core/tasty/TreePickler.scala

```diff
@@ -426,6 +426,13 @@ class TreePickler(pickler: TastyPickler) {
             writeByte(THROW)
             pickleTree(args.head)
           }
+          else if fun.symbol.originalSignaturePolymorphic.exists then
+            writeByte(APPLYsigpoly)
+            withLength {
+              pickleTree(fun)
```

Comment:
Shouldn't `originalSignaturePolymorphic` be involved here, somehow?

Reply:
Like how?  The function Select will be pickled as a SELECTin with the name "invokeExact" and the "in" as MethodHandle, which during unpickling will pick the original method and that's why we have to fix it up.